beetsplug/lastgenre/__init__.py

```diff
@@ -397,12 +394,13 @@ def _get_genre(
 
         # Return with a combined or freshly fetched genre list.
         if new_genres:
-            suffix = "whitelist" if self.whitelist else "any"
-            label += f", {suffix}"
-
-            if keep_genres:
-                label = f"keep + {label}"
-            return self._combine_genres(keep_genres, new_genres), label
+            resolved_genres = self._combine_genres(keep_genres, new_genres)
+            if any(resolved_genres):
```

Comment:
I kine of like this since it reads like `if we have any worthwhile resolved genres`.

The alternative here would be that `_combine_genres` ensures that no empty string in the list is allowed (another list comprehension in there... :-/), then here a simple `if resolved_genres:` would suffice.

For me what counts is that reading through `_get_genre`, since it is the main magic of the plugin to be as comprehensive as possible.

Reply:
> The alternative here would be that _combine_genres ensures that no empty string in the list is allowed

Seeing that the last list comprehension in `_resolve_genres` is

```python
valid_tags = [x for x in tags if self._is_valid(x)]
```

Can we exclude empty tags in `_is_valid`?

Reply:
forget it. reworked.

Reply:
Also note that I moved this list comprehension to a separate function since I reintroduced an original useful behavior of `lastgenre`: 926c7330e05d29c1af9ee775e3042a0facf2e79b